### ISSUE TITLE
Implement linear regression

### ISSUE THREAD

**Issue by jithendaraa** [jithendaraa/visualise.ml#1]
The more of these features the better. Implement LR with one or more **optimisation** techniques with:
- randomly initialize data of m samples and n features. (could implement a function in js just like [sklearn.datasets.make_regression](https://scikit-learn.org/stable/modules/generated/sklearn.datasets.make_regression.html))
- option to change **hyperparameters like learning rate**
- option to choose over different **normalizations** like L1 or L2 norm, minmax scaling, standardizations and how the data changes on changing these options
- options to choose the **loss functions** like RMSE, MSE, Absolute error,etc(you can add more on your own) 
- **animate** to give a better visual feel
- generate live loss vs iteration **graphs** for different learning params

**Comment by ditikrushna** [comment]
I want to work on this linear regression So may I know bit more regarding technology stack ?

**Comment by jithendaraa** [comment]
React is preferred, we can maybe use some visualization library with good compatibility with react like Victory or react-vis. Other libraries are fine as well
